python/ray/autoscaler/updater.py

```diff
@@ -343,16 +361,22 @@ def remote_shell_command_str(self):
         return inner_str + " docker exec -it {} /bin/bash\n".format(
             self.docker_name)
 
-    def docker_expand_user(self, string):
-        if string.find("~") == 0:
+    def docker_expand_user(self, string, any_char=False):
+        user_pos = string.find("~")
+        if user_pos > -1:
             if self.home_dir is None:
                 self.home_dir = self.ssh_command_runner.run(
                     "docker exec {} env | grep HOME | cut -d'=' -f2".format(
                         self.docker_name),
                     with_output=True).decode("utf-8").strip()
-            return string.replace("~", self.home_dir)
-        else:
-            return string
+
+            if any_char:
+                return string.replace("~/", self.home_dir + "/")
+
+            elif not any_char and user_pos == 0:
+                return string.replace("~", self.home_dir, 1)
+
+        return string
```

Comment:
IMO, if this continues to require weird edge casing, I would push a PR to remove this.

Reply:
(but no action required here)

Reply:
I could create a separate method. Regardless, there won't be any additions to this.